pkg/client/testclient/testclient_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testclient
+
+import (
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+)
+
+func TestNewClient(t *testing.T) {
+	o := NewObjects(api.Scheme)
+	if err := AddObjectsFromPath("../../../examples/guestbook/frontend-service.json", o); err != nil {
+		t.Fatal(err)
+	}
+	client := &Fake{ReactFn: ObjectReaction(o, latest.RESTMapper)}
+	list, err := client.Services("test").List(labels.Everything())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(list.Items) != 1 {
+		t.Fatalf("unexpected list %#v", list)
+	}
+
+	// same result
```

Comment:
To prove that you can invoke the request multiple times.  I don't have a test for it here, but you can define multiple PodLists and they are returned in order (so you can emulate things changing over time).

----- Original Message -----
> > +
> > +func TestNewClient(t *testing.T) {
> > +	o := NewObjects(api.Scheme)
> > +	if err :=
> > AddObjectsFromPath("../../../examples/guestbook/frontend-service.json",
> > o); err != nil {
> > +		t.Fatal(err)
> > +	}
> > +	client := &Fake{ReactFn: ObjectReaction(o, latest.RESTMapper)}
> > +	list, err := client.Services("test").List(labels.Everything())
> > +	if err != nil {
> > +		t.Fatal(err)
> > +	}
> > +	if len(list.Items) != 1 {
> > +		t.Fatalf("unexpected list %#v", list)
> > +	}
> > +
> > +	// same result
> 
> Did not understand, why are we doing the same thing twice?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6493/files#r27901068